### PR TITLE
temporal: drop default queue cap + disable search_attributes by default; docs handoff

### PR DESCRIPTION
**Pull request by lchoquel**
## Summary

PR is the cumulative diff of `feature/Adapt-to-api` against `feature/post-pr933-followups` — 5 commits, three concerns:

**Temporal config defaults (commits `ab6a1e1d`, `82fe422d`, `6d6d8af8`, `eafb1163`):**
- Add `AsyncExecutionNotEnabledError` for the case where async dispatch is requested but Temporal is disabled.
- Drop the `[temporal.queue_options.temporal_task_queue]` overlay from the package default `pipelex/pipelex.toml` (rationale: shipping an opinionated overlay for the baseline's queue name creates an orphan overlay the moment a downstream renames it, and config merging cannot delete baseline keys).
- Flip `[temporal.search_attributes].enabled` from `true` to `false` by default — operators owning the namespace policy who can't grant search-attribute admin to the worker no longer need to override.
- Align downstream configs with the package default: drop the now-orphan `[temporal.queue_options.temporal_task_queue]` from `pipelex/kit/configs/pipelex.toml` and `.pipelex/pipelex.toml`. Add a config-shape regression test pinning the empty-overlay contract; remove the previously-misnamed test that passed only because it read from the dev override.

**Docs (commit `9ff0d332`):**
- Append 2026-05-27 Session log entry to `_for_api/TODOS.md` documenting the workspace-level adapt plan at `../wip/adapt-pipelex-api-to-feature-API-readiness-2.md`.

## Test plan

- [x] `make tb` clean (boot test for config TOML changes)
- [x] `make agent-check` clean (ruff, plxt, pyright 0 errors, mypy 0 issues)
- [x] `tests/unit/pipelex/temporal/test_make_worker_uses_profile.py` + new `test_shipping_default_queue_options_empty.py` green
- [ ] (Reviewer) Verify the new default behavior is acceptable: out-of-the-box installs that don't override `[temporal.queue_options]` now run the default worker with no Temporal server-side queue cap. Deployments wanting a cap must explicitly configure `[temporal.queue_options.<queue>]` for whatever name they picked for `default_task_queue`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pipelex/pipelex.toml [hunk 1]
```diff
@@ -501,18 +508,16 @@ non_retryable_error_types = [
...
 # for examples.
+#
+# The baseline ships no entries here. Each deployment is expected to add its
+# own ``[temporal.queue_options.<queue>]`` block keyed by whatever name it
+# chose for ``default_task_queue`` (and any extra queues referenced by
+# ``activity_queues``). Shipping an opinionated entry for the baseline's
+# ``temporal_task_queue`` name would create an orphan overlay the moment a
+# downstream override renames the queue — and config merging cannot delete
+# baseline keys, so that orphan would be unrecoverable from the override.
 [temporal.queue_options]
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply pipelex/pipelex.toml:519]: ✅ Fixed via `eafb1163` — aligned downstream configs with the package default rather than restoring the overlay. Rationale: shipping the overlay re-introduces the orphan-overlay drift the empty-default was designed to avoid (config merge cannot delete baseline keys, so an overlay keyed on `temporal_task_queue` becomes unrecoverable if a deployment renames its default queue). Dropped `[temporal.queue_options.temporal_task_queue]` from `pipelex/kit/configs/pipelex.toml` and `.pipelex/pipelex.toml` so all three configs agree; added a config-shape regression test (`test_shipping_default_queue_options_empty.py`) pinning the empty-overlay contract. Out-of-the-box deployments now run the default worker with no Temporal server-side queue cap unless they explicitly configure one — this is the new documented contract.

[the remark this answers is not in the document]
Comment (lchoquel) [reply pipelex/pipelex.toml:519]: ✅ Fixed via `eafb1163` — this was the right diagnosis. Dropped `[temporal.queue_options.temporal_task_queue]` from both `pipelex/kit/configs/pipelex.toml` and `.pipelex/pipelex.toml`. All three configs now agree: no queue-specific overlay shipped anywhere. New regression test `test_shipping_default_queue_options_empty.py` loads `pipelex/pipelex.toml` directly and pins the empty-overlay shape; would catch a future re-add.

[the remark this answers is not in the document]
Comment (lchoquel) [reply pipelex/pipelex.toml:519]: ✅ Fixed via `eafb1163`. Note on the regression test you cited: `test_shipping_default_temporal_task_queue_has_1000_rate_cap` was passing under false pretenses — it loaded `.pipelex/pipelex.toml` (the dev override) at test time, not the shipping package default it claimed to guard. Replaced it with a config-shape test (`test_shipping_default_queue_options_empty.py`) that loads `pipelex/pipelex.toml` directly via `tomli` and asserts no queue-specific sub-table is shipped — pins the new design intent. The wiring case (`no overlay → cap=None`) is already covered by `test_max_task_queue_rate_cap_omitted_when_no_queue_options`. Aligned downstream configs (kit + dev override) with the package default rather than restoring the cap, to avoid the orphan-overlay drift the empty default exists to prevent.

File: .pipelex/pipelex.toml [hunk 2]
```diff
@@ -224,7 +224,7 @@ is_enabled = false
 # the operator owns the namespace policy and can't grant search-attribute
 # admin to the worker.
 [temporal.search_attributes]
-enabled = true
+enabled = false
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply .pipelex/pipelex.toml:227]: ✅ Fixed — updated the PR title and body to reflect the cumulative diff against `feature/post-pr933-followups`. The PR now explicitly documents the three concerns: `AsyncExecutionNotEnabledError`, the temporal config default changes (queue_options + search_attributes), and the docs Session log entry. The original `docs(plan)` framing covered only the last commit; the new title and Summary block list all five commits and their actual scope.
